File: LEETCODE/ARRAY/majority_element_1.py
class Solution(object):
    def majorityElement(self, nums):
        # Using Moores voting algorithm
        count =0
        element=0
        
        for val in nums:
            if count==0:
                element = val
            if element == val:
                count+=1
            else:
                count-=1
        return 

File: LEETCODE/ARRAY/majority_element_1.java
class Solution {
    public int majorityElement(int[] nums) {
        int count =0;
        int element = 0;
        
        for (int val:nums){
            if (count == 0){
                element = val;
            }
            if (element == val){
                count+=1;
            }
            else{
                count-=1;
            }
        }
        return element;
        
    }
}
